client/src/pages/Profile.js

```diff
@@ -1,13 +1,214 @@
+
+// ASSETS
+import investor from '../assets/images/investor.png';
+import linkedin from '../assets/images/linkedin-icon.png';
+import angellist from '../assets/images/angellist-icon.png';
+import projects from '../data/testing/projects';
+
+// CONTEXT
+import { userState } from '../provider/UserContext';
+
+const useStyles = makeStyles((theme) => ({
+  avatar: {
+    height: '100px',
+    margin: theme.spacing(4, 0, 2, 0),
+    width: '100px',
+  },
+  container: {
+    display: 'flex',
+    minHeight: 'calc(100vh - 64px)',
+    position: 'relative',
+  },
+  description: {
+    fontWeight: '500',
+    margin: theme.spacing(2, 0),
+    textAlign: 'center',
+    width: '70%',
+  },
+  location: {
+    color: '#999',
+    margin: theme.spacing(1, 0),
+  },
+  main: {
+    flex: 3,
+    padding: theme.spacing(8),
+  },
+  mainHeader: {
+    fontSize: '40px',
+    fontWeight: '600',
+    paddingLeft: '12px',
+  },
+  metaFooter: {
+    alignItems: 'center',
+    display: 'flex',
+    flexDirection: 'column',
+    padding: theme.spacing(2, 0),
+  },
+  userMeta: {
+    alignItems: 'center',
+    borderBottom: '1px solid #ddd',
+    display: 'flex',
+    flexDirection: 'column',
+    padding: theme.spacing(2, 0),
+  },
+  projects: {
+    margin: theme.spacing(8, 0),
+  },
+  sidebar: {
+    backgroundColor: '#fff',
+    flex: 1.3,
+  },
+  socialIcons: {
+    display: 'flex',
+    margin: theme.spacing(8, 0, 2, 0),
+  },
+  socialIcon: {
+    height: '40px',
+    margin: theme.spacing(0, 1),
+    width: '40px',
+  },
+  subheader: {
+    fontSize: '16px',
+    fontWeight: '600',
+    marginTop: theme.spacing(2),
+  },
+  textBubbles: {
+    display: 'flex',
+    flexWrap: 'wrap',
+    margin: theme.spacing(2, 0),
+  },
+  userName: {
+    fontSize: '24px',
+    fontWeight: '600',
+  },
+}));
+
+const Profile = () => {
+  const { state: { user: authUser, isAuthenticated }} = useContext(userState);
+  const { userId } = useParams();
+  const classes = useStyles();
+  const [currentUser, setCurrentUser] = useState({ _id: null });
+  const [ownUsersProfile, setOwnUsersProfile] = useState(false);
+  const [loading, setLoading] = useState(false);
+
+  const handleCurrentUser = (response) => {
+    if (isAuthenticated && authUser._id === response._id) {
+      setOwnUsersProfile(true);
+    } else {
+      setOwnUsersProfile(false);
+    }
+    setCurrentUser(response);
+    setLoading(false);
+  };
+
+  useEffect(() => {
+    const getUser = async () => {
+      setLoading(true);
+      fetch(`/api/users/${userId}`, {
+        method: 'GET',
+        headers: { 'Content-Type': 'application/json' },
+      })
+        .then((response) => response.json())
+        .then((response) => {
+          handleCurrentUser(response);
+        })
+        .catch((error) => {
+          setLoading(false);
+          console.log(error);
+        });
+    };
+    getUser();
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [userId]);

```

Comment:
We actually shouldn't have to get the user again if you are on the page of the current logged in user (since we have the user from the context)